python/fusion_engine_client/parsers/mixed_log_reader.py

```diff
@@ -627,6 +627,8 @@ def get_available_source_ids(self) -> Set[int]:
 
     def _populate_available_source_ids(self, num_messages_to_read: int = 10):
         self.available_source_ids = set()
+        stored_return_header = self.return_header
```

Comment:
What's the issue here, is it a recursion thing or something? I'd definitely recommend a comment explaining this.

Reply:
Nothing complicated, the lines in this function:
```python
                    result = self.read_next()
                    header = result[0]
```
assume the header is returned.